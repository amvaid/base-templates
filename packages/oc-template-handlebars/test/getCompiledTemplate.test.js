const fs = require('fs');
const { getCompiledTemplate } = require('../index.js');

test('Return compiled template when valid', () => {
  const template =
    'var oc=oc||{};oc.components=oc.components||{},oc.components.c6fcae4d23d07fd9a7e100508caf8119e998d7a9={compiler:[7,">= 4.0.0"],main:function(o,c,e,n,a){return"Hello world!"},useData:!0};';
  const key = 'c6fcae4d23d07fd9a7e100508caf8119e998d7a9';

  expect(getCompiledTemplate(template, key)).toMatchSnapshot();
});

test('Throw exception when js is not valid', () => {
  const template =
    'var oc=oc||{};oc.components=oc.components||{},oc.components.c6fcae4d23d07fd9a7e100508caf8119e998d7a9=nojavascript);';
  const key = 'c6fcae4d23d07fd9a7e100508caf8119e998d7a9';

  expect(() =>
    getCompiledTemplate(template, key)
  ).toThrowErrorMatchingSnapshot();
});

test('Be undefined when key is not valid', () => {
  const template =
    'var oc=oc||{};oc.components=oc.components||{},oc.components.c6fcae4d23d07fd9a7e100508caf8119e998d7a9={compiler:[7,">= 4.0.0"],main:function(o,c,e,n,a){return"Hello world!"},useData:!0};';
  const key = 'not valid key';

  expect(getCompiledTemplate(template, key)).toBeUndefined();
});
