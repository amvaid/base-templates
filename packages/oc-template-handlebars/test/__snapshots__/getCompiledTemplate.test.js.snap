// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Return compiled template when valid 1`] = `
Object {
  "compiler": Array [
    7,
    ">= 4.0.0",
  ],
  "main": [Function],
  "useData": true,
}
`;

exports[`Throw exception when not valid 1`] = `"received is not a function"`;
